### PR TITLE
improvement/config-save-spinner add spinner for submit button

### LINKED ISSUE #3406: Wizard: Add a proper UI handling for image restart on Cluster config update
### Which version of the app are you running?

[dac023e](https://github.com/provectus/kafka-ui/commit/dac023e)
the 'wizard' branch

### Is your proposal related to a problem?

On the update of cluster settings Kafka image is restarted and it is not reflected on the UI in any way.
There are also failed requests, that can be confusing or misleading:
image

### Describe the solution you'd like

Some handler on the UI with a load spinner or notification that image is being restarted and how long will it be unavailable.

### PR DESCRIPTION
**Pull request by NeiruBugz**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Resolves #3406
Add small spinner for button when pressing submit. I think it's the best decision for such case due to spinner visibility when user press submit
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/widgets/ClusterConfigForm/ClusterConfigForm.styled.ts [hunk 1]
```diff
@@ -60,3 +60,24 @@ export const FileUploadInputWrapper = styled.div`
   align-items: center;
   color: ${({ theme }) => theme.clusterConfigForm.fileInput.color}};
 `;
+
+export const Spinner = styled.div(
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/widgets/ClusterConfigForm/ClusterConfigForm.styled.ts:64]: i think maybe we can just make a general one cause i see this spinner code thing repeated across the application. we can make it customizable and just use that with `height` and `widths` and `colors` `props` etc and set the most common used one as a default value.

Comment (NeiruBugz) [reply kafka-ui-react-app/src/widgets/ClusterConfigForm/ClusterConfigForm.styled.ts:64]: Yeah, I thought about it too. Update PR soon, thanks!

Comment (NeiruBugz) [reply kafka-ui-react-app/src/widgets/ClusterConfigForm/ClusterConfigForm.styled.ts]: @Mgrdich @workshur updated spinner

File: kafka-ui-react-app/src/components/common/Button/Button.tsx [hunk 2]
```diff
@@ -20,5 +22,19 @@ export const Button: React.FC<Props> = ({ to, ...props }) => {
       </Link>
```
Comment (NeiruBugz) [inline kafka-ui-react-app/src/components/common/Button/Button.tsx:22]: Should I add spinner to Link button?

File: kafka-ui-react-app/src/components/common/Spinner/Spinner.styled.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+interface SpinnerProps {
+  width: number;
+  height: number;
+  borderWidth: number;
+  emptyBorderColor: boolean;
+  borderTheme?: string;
+  marginLeft?: number;
+}
+export const Spinner = styled.div<SpinnerProps>`
+  border-width: ${(props) => props.borderWidth}px;
```
Comment (NeiruBugz) [inline kafka-ui-react-app/src/components/common/Spinner/Spinner.styled.ts:12]: Split border properties for easier manipulating them

File: kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx [hunk 4]
```diff
@@ -1,10 +1,17 @@
 import React from 'react';
+import * as SP from 'components/common/Spinner/Spinner.styled';
 
 import * as S from './PageLoader.styled';
 
 const PageLoader: React.FC = () => (
   <S.Wrapper>
-    <S.Spinner role="progressbar" />
+    <SP.Spinner
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx:8]: We use S namespace for all styled components related to current component. 
```suggestion
    <Spinner
```

File: kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+      role="progressbar"
+      width={80}
+      height={80}
+      borderWidth={10}
+      emptyBorderColor={false}
```
Comment (workshur) [inline kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx:13]: do we really need to customize all this props here?

Comment (NeiruBugz) [reply kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx:13]: I think yes, but we can create default props object and customize on demand. In PageLoader it's gonna be empty in such case

File: kafka-ui-react-app/src/components/common/Button/Button.tsx [hunk 6]
```diff
@@ -20,5 +22,19 @@ export const Button: React.FC<Props> = ({ to, ...props }) => {
...
     );
   }
-  return <StyledButton type="button" {...props} />;
+  return (
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/common/Button/Button.tsx:25]: i think we should disable the button when it is in progress,
`<StyledButton type="button" {...props} disabled={props.disabled || props.inProgress}`

File: kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx (hunk 5 above)
Comment (NeiruBugz) [reply kafka-ui-react-app/src/components/common/PageLoader/PageLoader.tsx]: Updated in https://github.com/provectus/kafka-ui/pull/3551/commits/8a6706f0dd3c26aa2fc5f6bcd0ddf91625bf1092

File: kafka-ui-react-app/src/components/common/Button/Button.tsx (hunk 6 above)
Comment (NeiruBugz) [reply kafka-ui-react-app/src/components/common/Button/Button.tsx:25]: Updated + added test for this behavior

**Pull request merged**